### DISCUSSION TITLE
我发现apikey只要写死后被push到任何一个public项目，key就会被轮换。

### DISCUSSION THREAD

**Discussion by co2color** [ddiu8081/chatgpt-demo#265] [Q&A]
我github仓库有demo，我发现key只要被写进去并且提交代码了，key就会被轮换掉，请问这该怎么解决....因为我项目部署在vercel的，而vercel又是和github挂钩的。

现在的问题就是，一push代码，key就变了.....这个问题能解决吗

**Comment by ddiu8081** [comment]
Never commit `.env` to Git.

If you use Vercel, use `Environment Variables` on Vercel dashboard instead.

<img width="1010" alt="ScreenShot 2023-03-15 at 23 30 19@2x" src="https://user-images.githubusercontent.com/1998168/225359776-608c4f74-6971-4484-a4d8-6c1f8cc9080b.png">

**Comment by co2color** [comment]
> 我github仓库有demo，我发现key只要被写进去并且提交代码了，key就会被轮换掉，请问这该怎么解决....因为我项目部署在vercel的，而vercel又是和github挂钩的。
> 
> 现在的问题就是，一push代码，key就变了.....这个问题能解决吗

我试了试，仓库设置为私有就不会了....不知道是不是github账户和openai有什么绑定关系，我任何一个项目里面设置一个变量cont a = apikey....这样也会触发轮替...

**Reply by ddiu8081** [reply]
这是把敏感数据暴露到互联网了，OpenAI 或者 Github 检测到应该会自动帮你换掉key。
